dvc/command/gc.py

```diff
@@ -39,9 +62,9 @@ def run(self):
             return 1
 
         self.repo.gc(
-            all_branches=self.args.all_branches,
-            all_tags=self.args.all_tags,
-            all_commits=self.args.all_commits,
+            all_branches="branches" in exclude_scm_filters,
+            all_tags="tags" in exclude_scm_filters,
+            all_commits="commits" in exclude_scm_filters,
```

Comment:
Sorry, we didn't clarify it initially, but we need to adjust API as well, to have that same default behavior as CLI